python/rikai/contrib/torch/inspect/ssd.py

```diff
@@ -0,0 +1,191 @@
+#  Copyright 2022 Rikai Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Inspect detail of Torchvision SSD model
+
+https://pytorch.org/vision/stable/models.html#torchvision.models.detection.ssd300_vgg16
+
+"""
+
+from typing import Callable, Dict, List, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torchvision.models.detection.ssd import SSD
+from torchvision.ops.boxes import batched_nms, clip_boxes_to_image
+
+from rikai.contrib.torch.transforms.ssd import pre_processing
+from rikai.types import Box2d
+
+__all__ = [
+    "SSDClassScoresExtractor",
+    "class_scores_extractor_post_processing",
+    "class_scores_extractor_pre_processing",
+]
+
+
+class SSDClassScoresExtractor(torch.nn.Module):
+    """Extracts the scores (confidences) of each class for
+    all the detected bounding box.
+
+    Parameters
+    ----------
+    backend : :class:`torch.nn.Module`
+        The trained SSD model
+    topk_candidates : int, optional
+        The number of top candidates (classes) returned per box.
+
+    Returns
+    -------
+    [Dict[str, Tensor]]
+        With the form of
+        ``{"boxes": FloatTensor[N, 4], "labels": Int64Tensor[N, k], "scores": Tensor[N, k]}``
+    """  # noqa: E501
+
+    # SCHEMA to be used in Rikai SQL ML
+    SCHEMA = (
```

Comment:
is this used anywhere other than testing

Reply:
This one is used to be registered into the model catalog.